ide-backend/IdeSession/State.hs

```diff
@@ -257,3 +262,21 @@ managedData   :: Accessor ManagedFilesInternal [ManagedFile]
 
 managedSource = accessor _managedSource $ \x s -> s { _managedSource = x }
 managedData   = accessor _managedData   $ \x s -> s { _managedData   = x }
+
+{------------------------------------------------------------------------------
+  To allow for non server environments
+------------------------------------------------------------------------------}
+
+-- | fetch the directory that holds source files
+fetchSourceDir :: IdeStaticInfo -> FilePath
```

Comment:
Just a naming nitpick - I prefer `get` to `fetch`.  `fetch` doesn't seem to be used much in haskell APIs, and when it is, it tends to mean that we're fetching some data from an external resource.